homeassistant/components/systemmonitor/sensor.py

```diff
@@ -108,10 +155,12 @@ class SystemMonitorSensor(Entity):
     def __init__(self, sensor_type, argument=""):
         """Initialize the sensor."""
         self._name = "{} {}".format(SENSOR_TYPES[sensor_type][0], argument)
+        self.entity_id = ENTITY_ID_FORMAT.format(slugify(sensor_type + "_" + argument))
```

Comment:
Can we set a unique_id for the entity instead so we don't need to set entity_id explicitly? I'm assuming we do that now so that the entity_id won't change.

Reply:
```suggestion
        self._unique_id = ENTITY_ID_FORMAT.format(slugify(sensor_type + "_" + argument))
```

and then add:

    @property
    def unique_id(self):
        """Return the unique id."""
        return self._unique_id

as shown here: https://github.com/home-assistant/core/blob/214aa3de088811cee5aa102d4da5eef6592c29a5/homeassistant/components/nexia/entity.py#L31-L34

@MartinHjelmare will that be ok?

Reply:
No, the `unique_id` doesn't need the same format as the `entity_id`. It should be a string that is stable and unique across all entities for this platform. We don't need to prefix it with `sensor.`

Reply:
@MartinHjelmare I had originally attempted to only set a unique ID, but I thought it never worked while still preventing changes of the entity ID (and not breaking existing setups). I just tried it again and it turns out, that it seems to have been my entity registry file that contained some problematic entries from my earlier tests where still the name rather than the technical key (now used as the unique ID) was utilized.